bendev/device.py

```diff
@@ -179,7 +258,80 @@ def query(self, command, timeout=0, read_interval=0.05):
         """
         self.write(command)
         return self.read(read_interval=read_interval, timeout=timeout)
-    
+
+    def check_scpi_error(self, timeout=0, read_interval=0.05):
+        """Checks the SCPI error queue for errors and raises a RuntimeError if
+        any are found. This function is called automatically after each 
+        command sent to the device. If you want to check the error queue 
+        without sending a command, you can use the following command:
+        >>> device.query(":SYST:ERR?")
+        """
+        error_count = self.query(":SYST:ERR:COUNT?", timeout=timeout, read_interval=read_interval)
+        if error_count != "0":
+            error = self.query(":SYST:ERR?").split(",")
+            raise SCPIError(int(error[0]), error[1])
+
+    def cmd(self, command, timeout=0, read_interval=0.05):
+        """Sends a (non-query) command to the device. The SCPI error queue is
+        checked for errors and if any are found, a SCPIError is raised.
+
+        Arguments:
+            command: string containing the command to send
+            timeout: number, time in seconds before raising a TimeoutError 
+                if no message is received. 0 or None means never time out. 
+                Default: 0
+            read_interval: number, time in seconds to sleep between attempts 
+                to read.
+                Default: 0.05
+        
+        Returns:
+            the device reply as a string
+        """
+        self.write(command)
+        try:
+            self.check_scpi_error(timeout=timeout, read_interval=read_interval)
+        except SCPIError as e:
+            raise e
+
+    def cmd_query(self,
```

Comment:
These functions are a bit of a deviation from the existing philosophy of the module, which is fine but should be considered carefully:
- None of the other commands assume any SCPI functions are implemented, these assume there are error checking commands. I think this is pretty safe because I doubt anyone uses this for something other than our devices, and these were the first commands implemented.
- the other functions are minimal core functions without side effects. These new functions are utility functions that do a few things including:
  - delegate to write and to query
  - check errors
  - interprets replies to a degree
  that's fine and useful, especially considering that error checking is difficult since it flows down the same limited exchange route that the actual replies do, and bad commands may or may not find a reply, so it is good to have a standard process to perform this test.
- can we consider whether there are better possible names for these functions? cmd is a little ambiguous, all these functions issue commands. As shorthand it's probably fine.

Reply:
I think they're worth including in bendev as I've implemented these functions multiple times in other modules, so it would be nice to have them centrally. IIRC all SCPI devices are required to support some basic commands, including the error checking, so there shouldn't be an issue there.

The commands make simple wrapper classes very easy to implement, e.g. https://github.com/BenthamInstruments/ProductionTools_S400_487/blob/main/s400_487/amplifier.py and mean all errors are perculated through.

Happy to change name, could be very explicit `write_and_check_for_errors` `query_and_check_for_errors` ?

Reply:
how about 
- write_check
- query_check

then to get the checked versions of the other commands you just append _check